pkg/kubelet/kubelet_pods.go

```diff
@@ -533,6 +533,10 @@ func (kl *Kubelet) getServiceEnvVarMap(ns string, enableServiceLinks bool) (map[
 // Make the environment variables for a pod in the given namespace.
 func (kl *Kubelet) makeEnvironmentVariables(pod *v1.Pod, container *v1.Container, podIP string) ([]kubecontainer.EnvVar, error) {
 	var result []kubecontainer.EnvVar
+    if pod.Spec.EnableServiceLinks == nil {
+        enableServiceLinks := v1.DefaultEnableServiceLinks
+        pod.Spec.EnableServiceLinks = &enableServiceLinks
```

Comment:
I don't think it's safe to modify the pod spec here. better to just use a local variable:
```go
enableServiceLinks := v1.DefaultEnableServiceLinks
if pod.Spec.EnableServiceLinks != nil {
  enableServiceLinks = *pod.Spec.EnableServiceLinks
}
```